purchase_sale_stock_inter_company/models/purchase_order.py

```diff
@@ -22,11 +22,18 @@ def _prepare_sale_order_data(
         )
         if delivery_address:
             new_order.update({"partner_shipping_id": delivery_address.id})
+
+        potential_warehouse = self.env["stock.warehouse"].search(
+            [("partner_id", "=", self.partner_id.id)]
```

Comment:
You forgot the last part of your PR description (belongs to dest company)
```suggestion
            [
            ("partner_id", "=", self.partner_id.id),
            ("company_id", "=", dest_company.id),
            ]
```